Enable Redirect Settings

F4bsi opened the pull request:
<!--
    Thank you for contributing!
-->

#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/andi34/photobooth/blob/dev/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "x" next to an item)
<!-- Example:
- [x] Bug fix
-->

- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [ ] Other, please explain:

<!--
    Please ensure your pull request is ready:

    - Please run 'yarn build' before submitting to have consistent formatting for both JavaScript & SCSS
    - Please run 'yarn eslint' to make sure there's no lint issues
-->

<!--
    The following is required for all pull requests:
-->

#### What changes did you make? (Give an overview)
I added the possibility to change the destination of the redirection if the user is not logged in for the main page.
Currently you wil lbe redirected to the login page (if the index page is protected) but I wanted to have the possibility to redirect them to the gallery only page. Maybe this is helpful for other people too.

Wasn't quite sure which type the setting should have. Guess it could also be a select setting. With the input you could redirect the user to almost anywhere, but the select would be easier for the admin...

Hope I did everything right with the new structure and stuff :)

#### Is there anything you'd like reviewers to focus on?

F4bsi committed:
Enable Redirect Settings

andi34 commented on lib/configsetup.inc.php:
```diff
@@ -1629,6 +1629,13 @@
             'name' => 'protect[localhost_index]',
             'value' => $config['protect']['localhost_index'],
         ],
+        'index_redirect' => [
```
protect_index_redirect

andi34 commented on lib/configsetup.inc.php:
```diff
@@ -1629,6 +1629,13 @@
+        'index_redirect' => [
+            'view' => 'advanced',
+            'type' => 'input',
+	        'placeholder' => $defaultConfig['protect']['index_redirect'],
```
Space

F4bsi replied:
Hmm I don't see that in my ide. Maybe because I used tabs?

F4bsi replied:
```suggestion
	      'placeholder' => $defaultConfig['protect']['index_redirect'],
```

andi34 commented on resources/lang/en.json:
```diff
@@ -10,6 +10,7 @@
   "authentication:login_username": "Username",
   "authentication:protect_admin": "Protect admin panel",
   "authentication:protect_index": "Protect start screen",
+  "authentication:index_redirect": "Destination from start screen when not logged in",
```
:protect_

andi34 reviewed (changes requested):
Great idea! There's some nit-picks, see the comments.
Redirect should also follow the config inside the `livechroma.php`:
https://github.com/andi34/photobooth/blob/9cbc41a5ea70d2f5126c58d9d12ce60dfa3ecdb0/livechroma.php#L29


Maybe let's make it choosable instead an input?
```
        'protect_index_redirect' => [
            'view' => 'advanced',
            'type' => 'select',
            'name' => 'protect[index_redirect]',
            'placeholder' => $defaultConfig['protect']['index_redirect'],
            'options' => [
                'login' => 'Login',
                'gallery.php' => 'Standalone Gallery',
                'slideshow' => 'Standalone Slideshow',
            ],
            'value' => $config['protect']['index_redirect'],
        ],
```
Or we need to have the possibility to redirect to a completely different page (e.g. `private/my_private.php` or a page somewhere on the internet)?

F4bsi commented:
Glad that you like my idea. Crazy how far this project has come over the last 2 years!

I'm not 100% sure about the best way for selecting the redirection. But I guess there would be far less errors if we do a choosable/select option and I would say that for 99% of the users that would be enough. 

Maybe  we could do that for advanced and a free input for experts? But then you would have two inputs for the same thing...

F4bsi committed:
Change to select / update wording / livechroma

- Added logic to livechroma.php page
- Changed Setting from input to select and added Login/gallery.php and slideshow
- Changed keys to follow naming convention

F4bsi commented on lib/configsetup.inc.php:
```diff
@@ -1629,6 +1629,18 @@
+                'gallery.php' => 'Standalone Gallery',
+                'slideshow' => 'Standalone Slideshow',
+            ],
+            'placeholder' => $defaultConfig['protect']['index_redirect'],
```
@andi34 do you still need the placeholder here, or is the first option always the default?

andi34 replied:
Not sure to be true, used the language section as base and adjusted it

andi34 commented:
Maybe as last option to choose set to `Private` which will redirect to `private/index.php` from there you'll be able to redirect where ever you want to?

andi34 commented:
Sidenote: if you run `yarn format` before commiting, the formatting will be adjusted if needed :)

andi34 commented on lib/configsetup.inc.php:
```diff
@@ -1629,6 +1629,18 @@
+        'protect_index_redirect' => [
+            'view' => 'advanced',
+            'type' => 'select',
+            'options' => [
```
```
            'options' => [
                'login' => 'Login',
                'gallery.php' => 'Standalone Gallery',
                'slideshow' => 'Standalone Slideshow',
                'private' => 'Private (private/index.php)',
            ],
```

andi34 reviewed (changes requested):
Should be good to go if private is added to options

F4bsi commented:
Alright :) 
Already added it in my local repo. Will push it tonight when I'm back home.

F4bsi committed:
Add private option to redirect setting
Can be used to totally customize the page the user is redirected to by doing another redirect or adding custom content

F4bsi commented:
Hey @andi34, 

I added the private option (had a little bit different wording, is that ok?)
Also not quite sure about the help text and title, but I guess that it would be easy to change.

andi34 commented:
Thanks! And yeah, text can be adjusted easily.
